src/TriggerBinding/SqlTriggerMetricsProvider.cs

```diff
@@ -99,7 +101,8 @@ private async Task<long> GetUnprocessedChangeCountAsync()
         private SqlCommand BuildGetUnprocessedChangesCommand(SqlConnection connection, SqlTransaction transaction, IReadOnlyList<(string name, string type)> primaryKeyColumns, int userTableId)
         {
             string leasesTableJoinCondition = string.Join(" AND ", primaryKeyColumns.Select(col => $"c.{col.name.AsBracketQuotedString()} = l.{col.name.AsBracketQuotedString()}"));
-            string leasesTableName = string.Format(CultureInfo.InvariantCulture, LeasesTableNameFormat, $"{this._userFunctionId}_{userTableId}");
+            string leasesTableName = String.IsNullOrEmpty(this._userDefinedLeasesTableName) ? string.Format(CultureInfo.InvariantCulture, LeasesTableNameFormat, $"{this._userFunctionId}_{userTableId}") :
```

Comment:
Could we move this up to a common location instead of having to calculate the name in each place separately? If not, then at least we should have a common helper function so the logic is guaranteed to be the same.

Reply:
Added a helper function since moving to a common location would require the connection to be opened in SqlTriggerListener to get the userTableId.